File: core/src/com/symbol/ecs/component/DirectionComponent.java
package com.symbol.ecs.component;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.utils.Pool;

public class DirectionComponent implements Component, Pool.Poolable {

    public boolean facingRight = true;
    public boolean facingUp = true;
    public boolean yFlip = false;

    @Override
    public void reset() {
        facingRight = true;
        facingUp = true;
        yFlip = false;
    }

}

File: core/src/com/symbol/ecs/component/DirectionComponent.kt
package com.symbol.ecs.component

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.utils.Pool

class DirectionComponent : Component, Pool.Poolable {

    var facingRight = true
    var facingUp = true
    var yFlip = false

    override fun reset() {
        facingRight = true
        facingUp = true
        yFlip = false
    }

}

enum class Direction {

    Up,
    Down,
    Left,
    Right

}